tests/test_scada.py

```diff
@@ -1,87 +1,105 @@
+                                      start=self.start,
+                                      end=self.end,
+                                      fill_gaps='forwardfill',
+                                      down_sampling=True,
+                                      every_nth_value=2,
+                                      query_type_lab=False,)
 
-    # Compare average for each second value:
-    for ind, i in enumerate([0, 2, 4]):
-        assert np.isclose(np.mean(df_all[i:i + 2]), df['SomeParameter'][ind]), 'Averaging is incorrect.'
+        assert np.all(df_all[::2] == df), 'Downsampling did not return the correct values.'
 
-    # Testing lab query type:
-    df = sc.get_scada_values(parameters,
-                             start=start,
-                             end=end,
-                             query_type_lab=True,)
+        df = self.sc.get_scada_values(parameters,
+                                      start=self.start,
+                                      end=self.end,
+                                      fill_gaps='forwardfill',
+                                      every_nth_value=2,
+                                      query_type_lab=False,)
 
-    assert np.all(df['SomeParameter'] // 1 == -96), 'Not all values are correct for query type lab.'
+        # Compare average for each second value:
+        for ind, i in enumerate([0, 2, 4]):
+            is_correct = np.isclose(np.mean(df_all[i:i + 2]), df['SomeParameter'][ind])
+            assert is_correct, 'Averaging is incorrect.'
 
+        # Testing lab query type:
+        df = self.sc.get_scada_values(parameters,
+                                      start=self.start,
+                                      end=self.end,
+                                      query_type_lab=True,)
+        is_sorrect = np.all(df['SomeParameter'] // 1 == -96)
+        assert is_sorrect, 'Not all values are correct for query type lab.'
 
-def test_average_scada():
-    t = np.arange(0, 100, 10)
-    t_t, t_a = straxen.scada._average_scada(t / 1e9, t, 1)
-    assert len(t_a) == len(t), 'Scada deleted some of my 10 datapoints!'
+    @staticmethod
+    def test_average_scada():
+        t = np.arange(0, 100, 10)
+        t_t, t_a = straxen.scada._average_scada(t / 1e9, t, 1)
+        assert len(t_a) == len(t), 'Scada deleted some of my 10 datapoints!'
+
+    def resources_available(self):
+        """
+        Exception to skip test if external requirements are not met. Otherwise define 
+        Scada interface as self.sc.
+        """
+        if not straxen.utilix_is_configured('scada','scdata_url',):
+            self.skipTest("Cannot test scada since we have no access to xenon secrets.)")
+        
+        try:
+            self.sc = straxen.SCADAInterface(use_progress_bar=False)
+            self.sc.get_new_token()
+        except requests.exceptions.SSLError:
+            self.skipTest("Cannot reach database since HTTPs certifcate expired.")
```

Comment:
@jmosbacher does the same as what you are suggesting. The only problem is that not each query triggers the SSLError for some reason. In a few cases you get back a valid token to query parameters. I do not know why, but this I cannot fix on the client side.

Reply:
(But in a later query step the server tells you that the certificate has expired.)

Reply:
This is exactly the issue right:
https://github.com/XENONnT/straxen/pull/754#issuecomment-966905978

Reply:
Yes, this is why I put a skip flag at the very beginning of the test class so at the moment they are always skipped until the web API is fixed.